app/jobs/nomad_cleanup.py

```diff
@@ -0,0 +1,38 @@
+import nomad
+from app import Q
+from app.utils.dns import discover_service
+from datetime import timedelta
+import os
+
+from app.models import Tunnel
+
+
+@Q.job(func_or_queue="nomad", timeout=60000)
+def cleanup_old_nomad_box(job_id):
+    if os.getenv("FLASK_ENV") == "production":
+        nomad_client = nomad.Nomad(discover_service("nomad").url)
+    else:
+        nomad_client = nomad.Nomad(host=os.getenv("SEA_HOST", "0.0.0.0"))
+
+    try:
+        del_tunnel_nomad(nomad_client, job_id)
+    except nomad.api.exceptions.BaseNomadException:
+        cleanup_old_nomad_box.schedule(timedelta(hours=2), job_id, timeout=60000)
+
+
+def del_tunnel_nomad(nomad_client, job_id):
+    nomad_client.job.deregister_job(job_id)
+
+
+@Q.job(func_or_queue="nomad", timeout=100000)
+def check_all_boxes():
+    if os.getenv("FLASK_ENV") == "production":
+        nomad_client = nomad.Nomad(discover_service("nomad").url)
+    else:
+        nomad_client = nomad.Nomad(host=os.getenv("SEA_HOST", "0.0.0.0"))
+    deployments = nomad_client.job.get_deployments("ssh-client")
+
+    for deployment in deployments:
+        tunnel_exist = Tunnel.query.filter_by(job_id=deployment).first()
+        if not tunnel_exist:
+            cleanup_old_nomad_box.queue(deployment)
```

Comment:
I don't think this needs to be a separate job.  I'm _pretty sure_ that RQ supports just calling this directly instead of passing it through the queue.   Jobs that Queue Jobs are a code smell in my book.

Reply:
I changed it to just call the function directly.